ibm/service/cos/resource_ibm_cos_backup_vault_test.go

```diff
@@ -0,0 +1,337 @@
+					resource.TestCheckResourceAttr("ibm_cos_backup_vault.backup-vault", "activity_tracking_management_events", "true"),
+					resource.TestCheckResourceAttr("ibm_cos_backup_vault.backup-vault", "metrics_monitoring_usage_metrics", "true"),
+					resource.TestCheckResourceAttr("ibm_cos_backup_vault.backup-vault", "kms_key_crn", acc.KmsKeyCrn),
+				),
+			},
+		},
+	})
+}
+
+// tests
+
+func testAccCheckIBMCosBackup_Vault_Create_Valid(instance_id string, bucketVaultName string, region string) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+	}
+		
+	`, bucketVaultName, instance_id, region)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_Activity_Tracking_Valid(instance_id string, bucketVaultName string, region string, managementEvents bool) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		activity_tracking_management_events = "%t"
+	}
+		
+	`, bucketVaultName, instance_id, region, managementEvents)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_Metrics_Monitoring_Valid(instance_id string, bucketVaultName string, region string, usageMetrics bool) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		metrics_monitoring_usage_metrics = "%t"
+	}
+		
+	`, bucketVaultName, instance_id, region, usageMetrics)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_KP_Valid(instance_id string, bucketVaultName string, region string, kmsKeyCrn string) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		kms_key_crn = "%s"
+	}
+		
+	`, bucketVaultName, instance_id, region, kmsKeyCrn)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_AT_Enabled_MM_Disabled_Valid(instance_id string, bucketVaultName string, region string, managementEvents bool, usageMetrics bool) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		activity_tracking_management_events = "%t"
+		metrics_monitoring_usage_metrics = "%t"
+	}
+		
+	`, bucketVaultName, instance_id, region, managementEvents, usageMetrics)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_AT_Disabled_MM_Enabled_Valid(instance_id string, bucketVaultName string, region string, managementEvents bool, usageMetrics bool) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		activity_tracking_management_events = "%t"
+		metrics_monitoring_usage_metrics = "%t"
+	}
+		
```

Comment:
Both disabled and both enabled tests can be added here

Reply:
Both disabled case is the same as basic , because the terraform internally sets it as false.
Both enabled test case is already added
`TestAccIBMCosBackup_Vault_Create_All_Configurations_Valid`